internal/network/websocket.go

```diff
@@ -10,128 +10,175 @@ import (
 	"net/http"
 	"time"
 
+	kitlog "github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/gorilla/websocket"
 	"go.cryptoscope.co/muxrpc/v2"
 )
 
-func websockHandler(n *node) http.HandlerFunc {
+// WebsockHandler returns a "middleware" like thing that is able to upgrade a websocket request to a muxrpc connection and authenticate using shs.
+// It calls the next handler if it fails to upgrade the connection to websocket.
+// However, it will error on the request and not call the passed handler
+// if the websocket upgrade is successfull.
+func (n *node) WebsockHandler(next http.Handler) http.Handler {
 	var upgrader = websocket.Upgrader{
 		ReadBufferSize:  1024 * 4,
 		WriteBufferSize: 1024 * 4,
+
 		CheckOrigin: func(_ *http.Request) bool {
 			return true
 		},
+
+		// 99% of the traffic will be ciphertext which is impossible to distingish from randomness and thus also hard to compress
```

Comment:
```suggestion
		// 99% of the traffic will be ciphertext which is impossible to distingish
		// from randomness and thus also hard to compress
```

Reply:
i fixed the typo in distinguish, in addition to andre's suggestion